### PR TITLE
Fixing some uninitialized Txs

### PR DESCRIPTION
**Pull request by abi87**
Uninitialized means they will report empty ID for instance. I really think we don't want that

### CODE REVIEW

File: service/backend/cchainatomictx/construction.go [hunk 1]
```diff
@@ -255,11 +254,7 @@ func (b *Backend) parsePayloadTxFromString(transaction string) (*common.RosettaT
 		return nil, errUndecodableTx
 	}
 
-	if payloadsTx.Tx == nil {
-		return nil, errNoTxGiven
-	}
```
Comment (abi87) [inline service/backend/cchainatomictx/construction.go:260]: move to common.AvaxTx.Initialize implementation

File: service/backend/cchainatomictx/types.go [hunk 2]
```diff
@@ -33,27 +43,22 @@ func (c *cAtomicTx) Unmarshal(bytes []byte) error {
...
 	}
 
 	hash := hashing.ComputeHash256(c.Tx.Bytes())
 	return hash, nil
 }
 
-func (c *cAtomicTx) Hash() (ids.ID, error) {
-	if err := c.Tx.Sign(c.Codec, nil); err != nil {
-		return ids.Empty, err
-	}
```
Comment (abi87) [inline service/backend/cchainatomictx/types.go:55]: I checked that we only call this Hash on initilized txs

File: service/backend/pchain/construction.go [hunk 3]
```diff
@@ -329,9 +328,5 @@ func (b *Backend) parsePayloadTxFromString(transaction string) (*common.RosettaT
 		return nil, errUndecodableTx
 	}
 
-	if payloadsTx.Tx == nil {
-		return nil, errNoTxGiven
-	}
```
Comment (abi87) [inline service/backend/pchain/construction.go:334]: move to common.AvaxTx.Initialize implementation

File: service/backend/cchainatomictx/types.go [hunk 4]
```diff
@@ -33,27 +44,17 @@ func (c *cAtomicTx) Unmarshal(bytes []byte) error {
...
-}
 
-func (c *cAtomicTx) SigningPayload() ([]byte, error) {
-	if err := c.Tx.Sign(c.Codec, nil); err != nil {
-		return nil, err
-	}
+	return c.Initialize()
+}
 
-	hash := hashing.ComputeHash256(c.Tx.Bytes())
```
Comment (abi87) [inline service/backend/cchainatomictx/types.go:48]: I checked that we only call this SigningPayload on initilized txs

File: service/backend/pchain/types.go [hunk 5]
```diff
@@ -46,27 +55,17 @@ func (p *pTx) Unmarshal(bytes []byte) error {
...
 	}
-	if err := tx.Sign(p.Codec, nil); err != nil {
-		return err
-	}
 	p.Tx = &tx
-	return nil
-}
 
-func (p *pTx) SigningPayload() ([]byte, error) {
-	if err := p.Tx.Sign(p.Codec, nil); err != nil {
```
Comment (abi87) [inline service/backend/pchain/types.go:57]: I checked that we only call this SigningPayload on initilized txs

File: mapper/cchainatomictx/tx_builder.go [hunk 6]
```diff
@@ -27,12 +27,13 @@ func BuildTx(opType string, matches []*parser.Match, metadata Metadata, codec co
 	case mapper.OpExport:
 		return buildExportTx(matches, metadata, codec, avaxAssetID)
 	case mapper.OpImport:
-		return buildImportTx(matches, metadata, avaxAssetID)
+		return buildImportTx(matches, metadata, codec, avaxAssetID)
```
Comment (rajranjan0608) [inline mapper/cchainatomictx/tx_builder.go:30]: We can safely remove dummy tx signing from [estimateGas function](https://github.com/ava-labs/avalanche-rosetta/blob/pchain_construction_cleanup/service/backend/cchainatomictx/construction.go#L115L118).

Comment (abi87) [reply mapper/cchainatomictx/tx_builder.go:30]: Thanks for spotting this, cleaned up. Also dropped another duplicated signing.

**Pull request merged**